specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/examples/amlFilesystems_CreateOrUpdate.json

```diff
@@ -0,0 +1,144 @@
+{
+  "parameters": {
+    "resourceGroupName": "scgroup",
+    "amlFilesystemName": "fs1",
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "api-version": "2020-07-23-preview",
+    "amlFilesystem": {
+      "tags": {
+        "Dept": "ContosoAds"
+      },
+      "location": "westus",
+      "properties": {
+        "storageCapacityTiB": 3072,
+        "availabilityZones" : ["1", "2"],
+        "subnet": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/scgroup/providers/Microsoft.Network/virtualNetworks/scvnet/subnets/sub1",
+        "encryptionSettings": {
+          "keyEncryptionKey": {
+            "keyUrl": "https://examplekv.vault.azure.net/keys/kvk/3540a47df75541378d3518c6a4bdf5af",
+            "sourceVault": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/some-rg/providers/Microsoft.KeyVault/vaults/examplekv"
+          }
+        },
+        "maintenanceWindow" : "2020-09-13T13:15Z",
+        "HSMSettings" : {
+          "container" : "https://somecontainer.blob.core.windows.net/containername",
+          "containerKeyVaultKey" : {
```

Comment:
These appear to be called "secrets" in the portal.  So "containerkeyVaultSecret" might be more appropriate.

Reply:
There's keys and secrets.  
I based on this on the cache's encryption settings which seems to be a key vault key, not a secret.
I said Key Vault Key.
In the portal under key there is a Key Identifier which matches the KeyURL here.  I guess that word keyUrl could be switched to keyIdentifier.

For sourceVault, the description says resource id, so I went under properties of the key vault and used the resource id for the example.

"CacheEncryptionSettings": {
      "description": "Cache encryption settings.",
      "properties": {
        "keyEncryptionKey": {
          "x-ms-secret": true,
          "$ref": "#/definitions/KeyVaultKeyReference",
          "description": "Specifies the location of the key encryption key in Key Vault."
        }
      }
    },

"KeyVaultKeyReference": {
      "description": "Describes a reference to Key Vault Key.",
      "properties": {
        "keyUrl": {
          "type": "string",
          "description": "The URL referencing a key encryption key in Key Vault."
        },
        "sourceVault": {
          "description": "Describes a resource Id to source Key Vault.",
          "properties": {
            "id": {
              "type": "string",
              "description": "Resource Id."
            }
          }
        }
      },

Reply:
When I look at what CRP does for disks (DiskEncryptionSettings), they have two parameters. One is an encryption key (KV secret), and the other is a KEK (KV key).

https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/createorupdate#diskencryptionsettings

Reply:
Do we need to provide one secret/key for import, and potentially another for exporting to a different container?

Reply:
Thanks for digging into this everyeon.

"x-ms-secret": true,  <<=== maybe that's the differentiator

As for Joe's comment, I would suggest that just add a second key if we end up supporting a second container at some point since both will sort of go together.